WcaOnRails/app/models/competition.rb

```diff
@@ -216,7 +227,7 @@ def end_date
   end
 
   private def dates_must_be_valid
-    if self.year == 0 && self.month == 0 && self.day == 0 && @endYear == 0 && self.endMonth == 0 && self.endDay == 0
+    if !isConfirmed && self.year == 0 && self.month == 0 && self.day == 0 && @endYear == 0 && self.endMonth == 0 && self.endDay == 0
```

Comment:
The method name doesn't really match what's going on anymore. Could you add `if: :isConfirmed?` to the validation rule just like you did with the other validations?